src/pxfw.cpp

```diff
@@ -0,0 +1,654 @@
+		return 1;
+	} else {
+		if ((dev->dev_ID == PLEXTOR_OLD)){
+			printf("%s: Not supported PLEXTOR drive!\n",argv[0]);
+			return 1;
+		}
+	}
+//	plextor_get_life(dev);
+	if (flags & FL_RESET) {
+		plextor_reboot (dev);
+		exit (0);
+	}
+
+	if (flags & FL_EEPROM) {
+		int eelen;
+//		unsigned char buf[1024];
+		plextor_read_eeprom(dev, &eelen);
+		if (flags & FL_EEOF) {
+		    eefile = fopen(eefname,"w");
+		    if (!eefile) {
+			printf("%s: Can't write file: %s\n",argv[0],eefname);
+			exit (2);
+		    }
+		    fwrite(dev->rd_buf, eelen, 1, eefile);
+		    fclose(eefile);
+		}
+		exit (0);
+	}
+	
+	if (flags & FL_TEST) {
+		for (unsigned int opcode=0x0000; opcode<0x0100; opcode++) {
+			printf("Trying OpCode: ");
+		        print_opcode((unsigned char)opcode);			
+			if (!op_blacklisted((unsigned char)opcode)) {			
+			        int err = custom_command(dev, (opcode & 0x00FF));
+				if (!err) printf ("OK");
+				else if (err==-1) printf ("command didn't passed to dev");
+				else if (err==0x52000) printf ("*** command not supported ***");
+				else print_sense(err);
+			} else {
+				printf("!!! BLACK LIST !!!");
+			}
+			printf("\n");
+		}
+		exit(0);
+	}
+
+	if (!(flags & FL_FWIF)) exit(3);
+
+	fwfile = fopen(fwfname, "r");
+	if (!fwfile) {
+		printf("%s: Can't open file: %s\n",argv[0],fwfname);
+		return 1;
+	}
+
+//	fwblk=4096;
+	const int FB=64;
+	const int HH=16;
+	fwfsz=fsize(fwfile);
+	if (!fread(dev->rd_buf, FB, 1, fwfile)) {
+		printf("%s: error reading file: %s\n", argv[0], fwfname);
+		return 1;		
+	}
+
+	printf("this dev FW size : %7d (%06X)\n",fwsz,fwsz);
+	printf("FW file size       : %7d (%06X)\n",fwfsz,fwfsz);
+	printf("First %d bytes of FW:\n", FB);
+
+	for (i=0; i<(FB/HH); i++){
+		printf("| ");
+		for (j=0; j<HH; j++) printf("%02X ",dev->rd_buf[i*HH+j] & 0xFF);
+		printf(" | ");
+		for (j=0; j<HH; j++)
+			if ((dev->rd_buf[i*HH+j] & 0xFF) > 0x20)
+				printf("%c",dev->rd_buf[i*HH+j] & 0xFF);
+			else
+				printf(" ");
+		printf(" |\n");
+	}
+	FW_convert_to_ID(&FW_dev_ID,(char*) dev->rd_buf,&crc_offs);
+//	if (FW_dev_ID & (PLEXTOR_PREMIUM | PLEXTOR_PREMIUM2))
+	if (FW_dev_ID & (PLEXTOR_5224 | PLEXTOR_4824))
+		fwblk = 16384;
+	else
+		fwblk = 4096;
+	if (flags & FL_BACKUP) {
+		printf("BackUp feature not implemented yet...\n");
+		return 1;
+		
+		
+		for (i=0; i<fwblk; i++) dev->rd_buf[i]=0;
+		i=0;
+		if ((err = fwblk_read(dev, i*fwblk, fwblk, 2))) {
+			printf("** error reading data\n");
+		};
+
+		printf("First %d bytes of FW:\n", FB);
+		for (i=0; i<(FB/HH); i++){
+			printf("| ");
+			for (j=0; j<HH; j++) printf("%02X ", dev->rd_buf[i*HH+j] & 0xFF);
+			printf(" | ");
+			for (j=0; j<HH; j++)
+				if ((dev->rd_buf[i*HH+j] & 0xFF) > 0x20)
+					printf("%c", dev->rd_buf[i*HH+j] & 0xFF);
+				else
+					printf(" ");
+			printf(" |\n");
+		}
+	}
+
+	fseek(fwfile,0,SEEK_SET);
+	last=0;
+//	printf("fwblocks = %d\n",fwblocks);
+	fwblocks=fwfsz/fwblk;
+	fwblocks_crc=(crc_offs+2)/fwblk_crc;
```

Comment:
Codacy found an issue: [Variable 'fwblocks_crc' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)